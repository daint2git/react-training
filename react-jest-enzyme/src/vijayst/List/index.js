import React from 'react'

export default class List extends React.Component {
  static defaultProps = {
    data: [],
  }

  render() {
    return (
      <table className="myClass">
        <thead>
          <tr>
            <th>Email</th>
          </tr>
        </thead>
        <tbody>
          {this.props.data.map((item, index) => (
            <tr key={index}>
              <td>{item}</td>
            </tr>
          ))}
        </tbody>
      </table>
    )
  }
}
