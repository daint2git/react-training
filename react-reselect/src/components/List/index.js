import React from 'react'

const List = ({ todos, onUpdateStatus }) => {
  console.log('render List')
  return (
    <ul>
      {todos.map(todo => (
        <li
          key={todo.id}
          style={{ textDecoration: todo.finished ? 'line-through' : 'underline' }}
          onClick={() => onUpdateStatus(todo.id)}
        >
          {todo.task}
        </li>
      ))}
    </ul>
  )
}

export default List
