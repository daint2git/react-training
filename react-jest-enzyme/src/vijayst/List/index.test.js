import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import List from './'

describe('List', () => {
  const data = ['Email 1', 'Email 2', 'Email 3']
  test('Snapshot', () => {
    const component = renderer.create(<List />)
    const json = component.toJSON()
    expect(json).toMatchSnapshot()
  })

  test('Snapshot with data', () => {
    const component = renderer.create(<List data={data} />)
    const json = component.toJSON()
    expect(json).toMatchSnapshot()
  })

  describe('data', () => {
    let wrapper

    beforeEach(() => {
      wrapper = shallow(<List data={data} />)
    })

    it('List renders table', () => {
      expect(wrapper.find('table').length).toEqual(1)
    })

    it('Class of rendered table', () => {
      expect(wrapper.find('table').hasClass('myClass')).toEqual(true)
    })

    it('List renders column', () => {
      const arr = wrapper.find('th')
      expect(arr.length).toEqual(1)
      expect(arr.first().text()).toEqual('Email')
    })

    it('List renders data', () => {
      const arr = wrapper.find('td')
      expect(arr.length).toEqual(3)
      expect(arr.at(0).text()).toEqual('Email 1')
      expect(arr.at(1).text()).toEqual('Email 2')
      expect(arr.at(2).text()).toEqual('Email 3')
    })
  })
})
