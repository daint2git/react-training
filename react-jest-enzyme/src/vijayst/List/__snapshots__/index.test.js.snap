// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Snapshot 1`] = `
<table
  className="myClass"
>
  <thead>
    <tr>
      <th>
        Email
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;

exports[`List Snapshot with data 1`] = `
<table
  className="myClass"
>
  <thead>
    <tr>
      <th>
        Email
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Email 1
      </td>
    </tr>
    <tr>
      <td>
        Email 2
      </td>
    </tr>
    <tr>
      <td>
        Email 3
      </td>
    </tr>
  </tbody>
</table>
`;
